### DISCUSSION TITLE
Lost APP_KEY

### DISCUSSION THREAD

**Discussion by arghyadipchak** [firefly-iii/firefly-iii#5069] [Firefly III] [answered]
I lost the APP_KEY i used to deploy my firefly instance on docker but I do have the database. I made a new instance with a new data database (made the same user as well) and compared it to the old one. Most of the tables seemed to be replacable with the old data including the users table. The only problem I found was with the journal_meta table as it has some hashed data in the hash column. What can i do about that? And is there any other table/data I missed that needs to be taken care of if I need to use a new APP_KEY?

**Comment by JC5** [comment, marked as answer]
The APP_KEY is no longer used to encrypt financial data in the database. You may have to reset your password and your session will be invalid (meaning you have to login again). Otherwise, nothing bad will happen.

**Reply by arghyadipchak** [reply]
I tried with a new key, but it gave HTTP 500 error. Then I tried with the default key given in .env.example and it worked just fined (most probably i forgot to change the key before and used the default one). Everything works fine right now, but I wish to know the cause of the HTTP error.

Docker logs:
[default_key.log](https://github.com/firefly-iii/firefly-iii/files/7148093/default_key.log)
[new_key.log](https://github.com/firefly-iii/firefly-iii/files/7148097/new_key.log)

**Reply by JC5** [reply]
Bad key length, I'm guessing. Was it 32 characters?

**Reply by arghyadipchak** [reply]
Oh sorry, it's my bad. I used 30 characters instead of 32, it's working fine now. Thanks for all the help and I really appreciate you guys for developing firefly-iii. It's really useful and helps me a ton.
